### PR TITLE
Eli 47 transfer link inspector settings to granite console

### PR DESCRIPTION
**Pull request by prosis6**
#52 
#47 - implemented page with configs (load & save working automatically). available on /apps/etoolbox-link-
inspector/options/options.html.  Needed implementation of internal integration with Link Inspector logic.
#37
#35

### CODE REVIEW

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+package com.exadel.etoolbox.linkinspector.core.servlets;
+
+/**
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java:3]: (cosmetic) Please observe this notice is put before the imports section here , but the same one is put _after_ the imports section in the previous class. Need to put them in the same place,

Do we need these notices at all? No objection from my side, but:
- this is not customary for the project. Should all the other people start adding theirs in all the other classes?
- the authorship is clearly seen from GIT history anyway

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            default:
+                result = "granite/ui/components/coral/foundation/form/textfield";
+                break;
+        }
+        return result;
+    }
+
+    private Map<String, List<AttributeDefinition>> getServiceSettings() {
+        Map<String, List<AttributeDefinition>> result = new HashMap<>();
+        for (LinkResolver linkResolver : linkResolvers) {
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java:102]: `linkResolvers` can be null so it is better to wrap in `CollectionUtils.emptyIfNull()`

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/ServletUtil.java [hunk 3]
```diff
@@ -68,4 +72,21 @@ public static void writeJsonResponse(SlingHttpServletResponse response, String j
             LOG.error("Failed to write json to response", e);
         }
     }
+
+    public static Resource createTab(ResourceResolver resolver, String path, String title, Collection<Resource> children) {
```
[the remark this answers is not in the document]
Comment (prosis6) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/ServletUtil.java:76]: moved to GraniteUtil class

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java (hunk 1 above)
Comment (prosis6) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java:3]: removed

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java (hunk 2 above)
Comment (prosis6) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/SettingsDataSourceServlet.java:102]: thanks, added NPE check

**Review by aliakseiTraihel — APPROVED** [review]

**Pull request merged**